### PR TITLE
Improve merge page

### PR DESCRIPTION
**Pull request by illicitonion**
- Make links look like links
- Prompt for commit message
- Add a merge button
- Notify on merge started
- Notify on error merging
- Show commits being merged

### CODE REVIEW

File: lib/pages/merge.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+var CommitListView = require("../components/commit-list-view");
 
 var TriggerMixin = require("../logic/triggers").TriggerMixin([
-    "pr_info"
+    "merge_error", "pr_info",
 ]);
 
+const squashRewritingHistory = "Squash rewriting history";
+const squashMerge = "Squash leaving PR closed not merged";
+const mergeWithoutSquash = "Merge without squashing";
```
Comment (kegsay) [inline lib/pages/merge.js]: `CONSTANTS_IN_ALL_CAPS_PLEASE`

File: lib/logic/controller.js [hunk 2]
```diff
@@ -177,34 +185,51 @@ Controller.prototype._get_diff_comments = function(data) {
...
+            opt_sha || pr.getSource().getSha(),
+            commitMessage
+        ))
+        .delay(1000)
+        .then(() => this.refresh())
+        .catch((err) => this._handleMergeError(err));
+};
+
+Controller.prototype._handleMergeError = function(err) {
+    console.log("Got merge error: %s", JSON.stringify(err));
```
Comment (kegsay) [inline lib/logic/controller.js]: `console.error` would be more appropriate.

File: lib/logic/controller.js [hunk 3]
```diff
@@ -177,34 +185,51 @@ Controller.prototype._get_diff_comments = function(data) {
...
+    this.dispatcher.dispatch(triggers.create("merge_error", {"error": "Error merging"}))
```
Comment (kegsay) [inline lib/logic/controller.js:214]: Not exposing more error information from `err`?

File: lib/pages/merge.js [hunk 4]
```diff
@@ -15,68 +20,127 @@ module.exports = React.createClass({displayName: 'MergePage',
...
-        commitMessage += "\n\nSquash-merged from pull request #" + pr.getId() +
-            " from " + pr.getSource().getUser().name + "/" + pr.getSource().getRef();
-        this.props.controller.squashMergeWithoutRewrite(pr, commitMessage);
+
+        if (event.target.textContent === squashMerge) {
+            commitMessage += "\n\nSquash-merged from pull request #" + pr.getId();
+        }
+        else if (event.target.textContent === mergeWithoutSquash) {
+            commitMessage = "Merge pull request #" + pr.getId() + " from " +
+                pr.getSource().getUser().name + "/" + pr.getSource().getRef();
```
Comment (kegsay) [inline lib/pages/merge.js]: Given we're using ES6 features, it _may_ be clearer to use template strings here. Could be fun!

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/template_strings

**Comment by kegsay** [comment]
Superficially it looks good. I have not tested this; can you give confirmation of what you have manually tested with this?

File: lib/pages/merge.js (hunk 1 above)
Comment (illicitonion) [inline lib/pages/merge.js]: Done

File: lib/logic/controller.js (hunk 2 above)
Comment (illicitonion) [inline lib/logic/controller.js]: Done

File: lib/logic/controller.js (hunk 3 above)
Comment (illicitonion) [inline lib/logic/controller.js:214]: Not for now :) At some point I will

File: lib/pages/merge.js (hunk 4 above)
Comment (illicitonion) [inline lib/pages/merge.js]: Done

**Comment by illicitonion** [comment]
I've manually tested all three buttons, changing commit messages, that commit messages are properly pre-populated, the "merging" state of the button, the "merged" state of the button, the "error merging" state of the button.

I even have some automated tests for some of them, which I'll send you shortly :)

File: lib/logic/controller.js (hunk 3 above)
Comment (illicitonion) [inline lib/logic/controller.js:214]: Done I say, so that GPR will ignore this thread. We should really fix that...

**Comment by kegsay** [comment]
LGTM

File: lib/pages/merge.js [hunk 5]
```diff
@@ -1,9 +1,14 @@
 var React = require("react");
+var CommitListView = require("../components/commit-list-view");
 
 var TriggerMixin = require("../logic/triggers").TriggerMixin([
-    "pr_info"
+    "merge_error", "pr_info",
 ]);
 
+const SQUASH_REWRITING_HISTORY = "Squash rewriting history";
+const SQUASH_MERGE = "Squash leaving PR closed not merged";
```
Comment (kegsay) [inline lib/pages/merge.js:9]: Test comment

**Pull request merged**